hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/AvatarRelationTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.hxi_extension.rest.api;
+
+import static org.junit.Assert.*;
+
+import java.net.http.HttpClient;
+
+import org.junit.Test;
+
+import org.alfresco.rest.framework.core.exceptions.NotFoundException;
+import org.alfresco.rest.framework.resource.content.BinaryResource;
+
+public class AvatarRelationTest
+{
+    private static final String AGENT_ID = "agent-id";
+
+    private final HttpClient httpClient = HttpClient.newHttpClient();
+    private final AvatarRelation avatarRelation = new AvatarRelation(httpClient);
+
+    @Test
+    public void shouldThrowNotFoundExceptionIfAvatarIdIsDifferentThanDefault()
+    {
+        // given
+        String avatarId = "avatarId";
+
+        // when
+        assertThrows(NotFoundException.class, () -> avatarRelation.readById(AGENT_ID, avatarId, null));
+    }
+
+    @Test
+    public void shouldReturnAvatarForDefaultId()
+    {
+        // given
+        String avatarId = "-default-";
+
+        // when
+        BinaryResource binaryResource = avatarRelation.readById(AGENT_ID, avatarId, null);
```

Comment:
This unit test will fail if Wikipedia goes offline. Probably not very likely, but I don't like the test depending on an external resource like that. IMO it's fine to either delete the test for now, or extract the downloading code and use a mock in this test. I prefer the second option.

Reply:
Updated.